packages/server/test/mockHelper.ts

```diff
@@ -0,0 +1,222 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { MaybeAsync } from '@/modules/shared/helpers/typeHelper'
+import { isArray, isFunction } from 'lodash'
+import mock from 'mock-require'
+import { ConditionalPick } from 'type-fest'
+
+export type MockedFunctionImplementation = (...args: any[]) => MaybeAsync<any>
```

Comment:
I hate mocking...

Reply:
Yea, I don't really see any problem with mocking/faking per se, cause you inevitably need to set a fake boundary when it comes to integration tests. It's all about how the mocking is set up and used (like you once said - mock only what you own), and this kind of JS module mocking is standard practice in the JS ecosystem